cicecore/cicedyn/general/ice_step_mod.F90

```diff
@@ -85,6 +103,26 @@ subroutine step_prep
       call grid_average_X2Y('S',uatm,grid_atm_dynu,uatmT,'T')
       call grid_average_X2Y('S',vatm,grid_atm_dynv,vatmT,'T')
 
+      !-----------------------------------------------------------------
+      ! Compute uvelT_icep, vvelT_icep
+      !-----------------------------------------------------------------
+
+      if (first_call) then
+         allocate(uvelT_icep(nx_block,ny_block,max_blocks))
+         allocate(vvelT_icep(nx_block,ny_block,max_blocks))
+         uvelT_icep = c0
+         vvelT_icep = c0
+      endif
+
+      call icepack_query_parameters(highfreq_out=highfreq)
+
+      if (highfreq) then
+         call grid_average_X2Y('A', uvel, 'U', uvelT_icep, 'T')
```

Comment:
what does 'icep' signify?

Reply:
Oh, it's probably for 'icepack'.  That wasn't intuitive to me.

Reply:
_icep means "passed into icepack".  This is a pretty awkward implementation.  [u,v]velT_icep are the ice velocities passed into icepack for computation of the wind stress.  If highfreq is true, then [u,v]velT_icep = [u,v]vel interpolated to the T grid.  If highfreq is false, then [u,v]velT_icep=0.  The implementation was refactored recently to make this clearer, but it's still awkward.  In my opinion, highfreq and [u,v]velT should probably be passed into icepack and then icepack should decide what to do.  Even better, it shouldn't be called highfreq.  The option should be called something like "compute_wind_stress_relative_to_ice_velocity".  It might mostly be used with highfreq coupling, but I think how this is setup is a bit confusing.